catalyst/core/engine.py

```diff
@@ -81,7 +93,12 @@ def optimizer_step(self, loss, model, optimizer) -> None:
 
     @abstractmethod
     def pack_checkpoint(
-        self, model=None, criterion=None, optimizer=None, scheduler=None, **kwargs,
+        self,
+        model: Model = None,
+        criterion: Criterion = None,
+        optimizer: Optimizer = None,
+        scheduler: Scheduler = None,
```

Comment:
should they be `RunnerModel`, `RunnerCriterion`, etc? from `catalyst.typing`

Reply:
I think that Model, Criterion, Optimizer, and Scheduler is a base blocks and engine should work only with pytorch primitives.

Reply:
why shouldn't it also work with `Dict[Model]` case? that's what `RunnerModel` is about